include/TPP/Passes.td

```diff
@@ -115,4 +115,78 @@ def TransformDropSchedulePass : Pass<"transform-drop-schedule", "ModuleOp"> {
   let constructor = "mlir::tpp::createTransformDropSchedulePass()";
 }
 
+def PackMatmul : Pass<"pack-matmul", "func::FuncOp"> {
+  let summary = "Convert matmul to block layout and back";
+  let description = [{
+    Block Matmul as: [NB][KB][nb][kb] += [NB][CB][nb][cb] * [KB][CB][cb][kb] If
+    the Matmul has a relu operation as its consumer block also the relu operation.
+  }];
+  let options = [
+    ListOption<"blockingFactors", "block-factors", "int64_t", 
+               "Blocking factor for relayout">
+  ];
+  let constructor = "mlir::tpp::createPackMatmulPass()";
+}
+
+def PackConv2DNchwFchw : Pass<"pack-conv2DNchwFchw", "func::FuncOp"> {
+  let summary = "Convert Conv2DNchwFchw to block layout and back";
+  let description = [{
+    Block Conv2DNchwFchw as: [N][BK][P][Q][bk] += [N][BC][H][W][bc] * [BK][BC][R][S][bk][bc]
+                             output            += image             * filter
+    Pack the image's channel with a block factor BC.
+    Pack the filter's channels C and K with a block factor of BC and BK.
+    Pack the output's channel K with a block factor BK.
+  }];
+  let options = [
+    ListOption<"blockingFactors", "block-factors", "int64_t",
+               "Blocking factor for relayout">
+  ];
+  let constructor = "mlir::tpp::createPackConv2DNchwFchwPass()";
+}
+
+def PackConv2DNhwcHwcf : Pass<"pack-conv2DNhwcHwcf", "func::FuncOp"> {
```

Comment:
Untested code

Reply:
the passes and the transform uses the same code. I would like to avoid duplicating tests.

Reply:
The passes are not being tested, even if the code that the pass uses is being tested. Since we're moving to passes, and will eventually delete the transforms from `main`, we need those tests to be in passes.